src/dotnet/CommandFactory/CommandUsingResolver.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.CommandFactory
+{
+    public static class CommandUsingResolver
```

Comment:
Good name? I want to call it CommandFactory2 in the beginning. This used to be on Command.cs

Reply:
CommandFactoryUsingResolver? I feel like your name may work for now though.